posthog/temporal/batch_exports/spmc.py

```diff
@@ -251,18 +251,12 @@ async def start(
 
         record_batches_count = 0
         records_count = 0
-        record_batch_generator = self.generate_record_batches_from_queue(queue, producer_task)
 
         await self.logger.adebug("Starting record batch writing loop")
 
-        writer._batch_export_file = writer.create_temporary_file()
-
-        while True:
-            try:
-                record_batch = await anext(record_batch_generator)
-            except StopAsyncIteration:
-                break
+        writer._batch_export_file = await asyncio.to_thread(writer.create_temporary_file)
```

Comment:
it's annoying Python doesn't have async support for writing temporary files

Reply:
Async file operations are not really a thing. I think the alternative would be to use a library like `aiofiles` but it's also using threads under the hood like here.